core/src/main/java/org/fao/geonet/kernel/DataManager.java

```diff
@@ -566,7 +568,7 @@ public void indexMetadata(final List<String> metadataIds) throws Exception {
     /**
      * TODO javadoc.
      */
-    public void indexMetadata(final String metadataId, boolean forceRefreshReaders) throws Exception {
+    public void indexMetadata(final String metadataId, boolean forceRefreshReaders, ISearchManager searchManager) throws Exception {
```

Comment:
I think would be better to keep a method also with `metadataId` and `forceRefreshReaders` only that calls this one sending `searchManager` as `null`, that is used in all places changed to send searchManager to `null`. Wth the change there's quite changes where `searchManager` is send as `null`.